twine/exceptions.py

```diff
@@ -29,7 +29,16 @@ class RedirectDetected(TwineException):
     redirecting them.
     """
 
-    pass
+    @classmethod
+    def from_args(cls, repository_url, redirect_url):
+        msg = "\n".join([
+            "Unexpected redirect from {0} to {1}."
+            .format(repository_url, redirect_url),
+            "You might need to change the configured repository URL.",
```

Comment:
In the more common cases a redirect here is innocuous and perhaps changing the URL is fine. But there could be cases where folks are being redirected to a malicious URL that will be able to:

- steal credentials
- store the uploaded package data

In that case, I don't want folks to think "Geeze, really?! I have to copy and paste this into my pypirc for this stupid tool to work?! WTF IS WRONG WITH THOSE PEOPLE" and then to angrily and carelessly copy and paste the malicious URL.

Granted, I think the this is very unlikely, but still I would like to see more cautious wording. Something like:

```
You should examine the URLs carefully and if you trust them, update your configured repository URL.
```

Also, would you share your reasoning for using a newline to separate these lines? Has the project moved towards preferring newlines?

Reply:
Thanks for the feedback. I wordsmithed the error message a bit in 1b80776. Re: newlines, I thought it was more readable, and there are a couple instances of them being used in the project. But I don't feel strongly, and am happy to change it.

https://github.com/pypa/twine/blob/da07f786a3633d8a126ecbccabebb2449c679215/twine/exceptions.py#L50-L60

https://github.com/pypa/twine/blob/da07f786a3633d8a126ecbccabebb2449c679215/twine/utils.py#L122-L132